pages/upgrade-guide/4.0.md

```diff
@@ -0,0 +1,186 @@
+* `jQuery.isNumeric`
+* `jQuery.isWindow`
+* `jQuery.fx.interval`
+* `jQuery.parseJSON`
+* `jQuery.nodeName`
+* `jQuery.now`
+* `jQuery.trim`
+* `jQuery.type`
+
+#### Breaking change: Remove undocumented Array methods
+
+jQuery 4.0 removes `push`, `sort`, and `splice` from the jQuery prototype. These methods were copies of the methods of the same names on the native Array prototype, but were never documented and were not intended for public use. Some plugins may have used them, so we are still marking this as a breaking change.
+
+#### Slim build breaking change: Remove `callbacks` and `deferred` modules
+
+The slim build of jQuery 4.0 no longer includes the `callbacks` and `deferred` modules. Use the full build if you need these modules, or use native Promise objects.
+
+#### Breaking change: Remove undocumented `root` parameter of `jQuery()`
+
+The `root` parameter of `jQuery()` was never documented and was only used internally. Also, it has not been necessary since jQuery 1.9 when `jQuery.sub()` was removed.
+
+#### Important fix: Add `exports` to package.json
+
+jQuery exports multiple builds that can be used in different environments. This includes the default build as both a universal module (UMD) and an ESM module, the slim build in UMD and ESM, and a factory build for running jQuery with DOM emulators like JSDOM. The `exports` field in `package.json` is used to specify which build is used in different environments. This change is not expected to affect existing code.
+
+### CSS
+
+#### Breaking change: `px` no longer automatically added to most unitless values
+
+jQuery 4.0 switches strategies when it comes to setting CSS properties that require units. Previously, jQuery would automatically add `px` to unitless values, with some exceptions. This behavior has been removed in jQuery 4.0.
+
+Instead, jQuery will now only add `px` to a limited set of properties that are known to require it. This change may affect existing code that relies on the old behavior.
+
+### Data
+
+#### Important fix: `Object.prototype` pollution
+
+jQuery 4.0 includes a fix to ensure event and data keys matching `Object.prototype` properties are supported without overriding the native properties. This change is not expected to affect existing code unless it specifically used native property names as event or data keys, which may have caused unexpected behavior anyway.
+
+### Deferred
+
+#### Breaking change: Removed jQuery.Deferred.getStackHook
+
+This has been renamed since 3.7.0 to jQuery.Deferred.getErrorHook. It is used to pass the original error to jQuery.Deferred.exceptionHook. Previously, we recommended passing the stack but that didn't play well with source maps.
+
+### Event
+
+#### Breaking change: Stop shimming focusin & focusout events
+
+jQuery 4.0 no longer shims the `focusin` and `focusout` events. This only affects code that relies on the order of these events as they relate to `focus` and `blur`. This change is expected to affect very few users.
```

Comment:
For the record, all non-IE browsers will now - when moving focus between inputs - trigger events in the order: `blur`, `focusout`, `focus`, `focusin`. In IE, because we shim `focus` & `blur` using `focusin` & `focusout` without caring to maintain a specific order (which we do in 3.x), `blur` can be before or after `focusout` and `focus` can be before or after `focusin`.

I'm not sure if it's worth documenting it here, especially the arcane IE aspects. If anything, we could consider just writing that non-IE browsers now fire `blur` before `focusout` and `focus` before `focusin` and mention that in IE the order can be different.